server/datastore/mysql/software_installers.go

```diff
@@ -1296,36 +1296,65 @@ WHERE
 }
 
 func (ds *Datastore) GetHostLastInstallData(ctx context.Context, hostID, installerID uint) (*fleet.HostLastInstallData, error) {
-	// TODO(uniq): I think that if there's none in host_software_installs, must take
-	// latest in upcoming queue (latest attempt might actually be in upcoming).
+	hostLastInstall, err := ds.getLatestUpcomingInstall(ctx, hostID, installerID)
+	if err != nil && errors.Is(err, sql.ErrNoRows) {
+		hostLastInstall, err = ds.getLatestPastInstall(ctx, hostID, installerID)
+	}
+
+	if err != nil && errors.Is(err, sql.ErrNoRows) {
+		return nil, nil
+	}
+
+	return hostLastInstall, err
+}
+
+func (ds *Datastore) getLatestUpcomingInstall(ctx context.Context, hostID, installerID uint) (*fleet.HostLastInstallData, error) {
+	var hostLastInstall fleet.HostLastInstallData
 	stmt := `
-		SELECT execution_id, hsi.status
-		FROM host_software_installs hsi
-		WHERE hsi.id = (
-			SELECT
-				MAX(id)
-			FROM host_software_installs
-			WHERE
-				software_installer_id = :installer_id AND host_id = :host_id
-			GROUP BY
-				host_id, software_installer_id)
-`
+SELECT 
+	execution_id, 
+	'pending_install' AS status
+FROM 
+	upcoming_activities
+WHERE 
+	id = (
```

Comment:
Should we use the LEFT OUTER JOIN approach for this, as it's supposed to be much faster? (https://stackoverflow.com/a/23285814)

Maybe we should wait until we have the tests fixed and all towards the end of the story to do this refactor (or not) as time permits. What do you think?